AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -208,6 +209,25 @@ - (void)removePlayerItemObservers:(AVPlayerItem *)playerItem
   [notificationCenter removeObserver:self name:AVPlayerItemNewErrorLogEntryNotification object:playerItem];
 }
 
+- (void)addPlayerObservers:(AVPlayer *)player
+{
+  if (player == nil) {
+    return;
+  }
+
+  [player addObserver:self forKeyPath:kRate options:NSKeyValueObservingOptionNew context:ASVideoNodeContext];
+}
+
+- (void) removePlayerObservers:(AVPlayer *)player
+{
+  @try {
+    [player removeObserver:self forKeyPath:kRate context:ASVideoNodeContext];
+  }
+  @catch (NSException * __unused exception) {
+    NSLog(@"Unnecessary KVO removal");
+  }
```

Comment:
@flovouin @Adlai-Holler It looks to me like this is really a separate issue from the playerStatus behaviour improvements, as like flovouin says this pattern is followed elsewhere in the code.

Is this exception handling something we could raise as a new bug and deal with afterwards?

My project is dependent on these changes you see, so I'm keen to see these in mainline or even better a release if possible. I'm more than happy to help with testing.